cmake/modules/AddPureSwift.cmake

```diff
@@ -280,12 +254,27 @@ function(add_pure_swift_host_library name)
         >)
   else()
     # Emit a swiftmodule in the current directory.
-    set_target_properties(${name} PROPERTIES
-        Swift_MODULE_NAME ${name}
-        Swift_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
-    set(module_file "${CMAKE_CURRENT_BINARY_DIR}/${name}.swiftmodule")
+    set(module_dir "${CMAKE_CURRENT_BINARY_DIR}/modules")
```

Comment:
Fragile modules are now created in `${CMAKE_CURRENT_BINARY_DIR}/modules`. This is to avoid PR testing failures in incremental bots. This PR changes `_CompilerRegexParser` to fragile, if we create `${CMAKE_CURRENT_BINARY_DIR}/_CompilerRegexParser.swiftmodule` , subsequent PR testings will be confused because `-I ${CMAKE_CURRENT_BINARY_DIR}` is added by CMake. Pre-patch https://github.com/apple/swift/pull/73799 can avoid that, but just in case.